### DISCUSSION TITLE
Is it possible to render <CustomComponent /> instead static value in cells ? 

### DISCUSSION THREAD

**Discussion by Steffi3rd** [TanStack/table#2057] [General]
Hi,

I would like to know if it's possible to do this :

```
const data = React.useMemo(
  () => [
    {
      col1: 'Hello',
      col2: <CustomComponent  id="hello"/>,
    },
    {
      col1: 'react-table',
      col2: <CustomComponent id="rocks"/>,
    },
  ],
  []
)
```

**Comment by fnn** [comment]
Hey,
you can do this:

```javascript
const data = React.useMemo(
  () => [
    {
      Header: 'Hello',
      accessor: 'hello',
      Cell: () => <CustomComponent  id="hello"/>
    },
  ],
  []
)
`

**Reply by Ajayghoshm** [reply]
This is for Column Array, right?
How is it possible for Row array?

**Reply by fnn** [reply]
Yes. This is for the columns.
Inside Cell you have access to the row data:
```javascript
Cell: ({ row }) => <Component prop={row.original.someRowProp} />
```
In your row data array is only the “raw data” you need.

**Comment by JeffEmery** [comment]
`data` provides the table with the actual data you are managing
`columns` is where you handle the rendering of cells based on that data
as @fnn describes, you want to use Cell **but on the columns array** not on the data array.
